### PR TITLE
iox-#51 Create custom action for tool build step

### LINKED ISSUE #51: Create custom actions for tool build step
## Brief feature description

The CI config to build cargo-nextest and grcov is copy&pasted. To reduce duplication, this should be refactored out to a custom action.

### PR DESCRIPTION
**Pull request by elBoberido**
## Pre-Review Checklist for the PR Author

1. [x] Code follows the Rust coding style and is formatted with `rustfmt`
1. [x] Branch follows the naming format (`iox-123-this-is-a-branch`)
1. [x] Commits messages are according to this [guideline][commit-guidelines]
    - [x] Commit messages have the issue ID (`iox-#42 commit text`)
    - [x] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [x] Update the PR title
   - Follow the same conventions as for commit messages
   - Link to the relevant issue
1. [x] Relevant issues are linked
1. [x] Add sensible notes for the reviewer
1. [x] All checks have passed (except `task-list-completed`)
1. [x] Assign PR to reviewer

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php

## Checklist for the PR Reviewer

- [ ] Commits are properly organized and messages are according to the guideline
- [ ] Code according to our coding style and naming conventions
- [ ] PR title describes the changes

## Post-review Checklist for the PR Author

1. [ ] All open points are addressed and tracked via issues

## References

- Closes #51

### CODE REVIEW

File: .github/actions/build-and-cache-rust-tool/action.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
+name: 'Build and cache random Rust tools'
```
Comment (dkroenke) [inline .github/actions/build-and-cache-rust-tool/action.yml:1]: I guess you want maybe to use `on.workflow_call` for creating a reusable workflow.
https://docs.github.com/en/actions/using-workflows/reusing-workflows#creating-a-reusable-workflow

```suggestion
on:
  workflow_call:
name: 'Build and cache random Rust tools'
```

Comment (elBoberido) [reply .github/actions/build-and-cache-rust-tool/action.yml:1]: Maybe. But after reading the error message more closely it tells me that the `shell` variable is missing.

Comment (elBoberido) [reply .github/actions/build-and-cache-rust-tool/action.yml:1]: Seems to work now. Just want to check if we get a problem on windows in the future

Comment (elBoberido) [reply .github/actions/build-and-cache-rust-tool/action.yml]: It seems there will be a problem on windows but not due to this refactoring but the path for the upload of the artifact

**Pull request merged**